persistence/aggregate.go

```diff
@@ -41,6 +59,14 @@ type AggregateRepository interface {
 		ctx context.Context,
 		hk, id string,
 	) (AggregateMetaData, error)
+
+	// LoadAggregateSnapshot loads the latest snapshot for an aggregate instance.
+	//
+	// hk is the aggregate handler's identity key, id is the instance ID.
+	LoadAggregateSnapshot(
+		ctx context.Context,
+		hk, id string,
+	) (*AggregateSnapshot, error)
```

Comment:
I'd avoid using a pointer here, because:
- we don't really want reference semantics, `AggregateSnapshot` is just a plain-old-value
- It's less clear to the caller that `nil` is meaningful (vs the `ok bool` style)
- marginally less GC pressure without the heap allocation

```suggestion
	) (AggregateSnapshot, bool, error)
```